instrumentation/http_client/lib/opentelemetry/instrumentation/http_client/patches/client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module HttpClient
+      module Patches
+        # Module to prepend to HTTPClient for instrumentation
+        module Client
+          private
+
+          def do_get_block(req, proxy, conn, &block) # rubocop:disable Metrics/AbcSize
+            uri = req.header.request_uri
+            url = "#{uri.scheme}://#{uri.host}"
+            request_method = req.header.request_method
+            attributes = {
+              'http.method' => request_method,
+              'http.scheme' => uri.scheme,
+              'http.target' => uri.path,
+              'http.url' => url,
+              'peer.hostname' => uri.host,
+              'peer.port' => uri.port
+            }.merge(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+
+            tracer.in_span("HTTP #{request_method}", attributes: attributes, kind: :client) do |span|
+              OpenTelemetry.propagation.inject(req.header)
+              super.tap do
+                response = conn.pop
+                annotate_span_with_response!(span, response)
+                conn.push response
+              end
+            end
+          end
+
+          def annotate_span_with_response!(span, response)
+            return unless response&.status_code
+
+            status_code = response&.status_code.to_i
```

Comment:
```suggestion
            status_code = response.status_code.to_i
```
No need to be lonely - you already tested it above, so we know it is non-`nil`.